crt_portal/cts_forms/views.py

```diff
@@ -271,6 +304,8 @@ def index_view(request):
         lambda text: Campaign.objects.filter(internal_name=text).first()
     )
 
+    _was_sent_constant_writer_email(requested_reports)
+
     final_data = {
         'form': Filters(request.GET),
         'profile_form': profile_form,
```

Comment:
to further illustrate the comment I made above, the usage here would be:

```suggestion
    final_data = {
        'form': Filters(request.GET),
        'profile_form': profile_form,
        'sent_templates': _get_sent_templates_for_reports(requested_reports),
```

and then you'd render from that dict based on the report id in the html.


A second thought: This approach also makes the list page dependent on the constant writer icon query finishing. You might want to unblock that using some javascript after the page loads, [similar to this pattern](https://github.com/usdoj-crt/crt-portal/blob/develop/crt_portal/static/js/show_count.js)

Reply:
Updated to use JS to run this query after page load